### ISSUE TITLE
Add support for processing wavs in memory without passing a file path string.

### ISSUE THREAD

**Issue by MJWeldy** [joeweiss/birdnetlib#79]
I have a number of scripts that would need to be able to process in memory wavs instead of wavs saved to disk so it would be nice to add an argument flag for passing the audio vector itself.

**Comment by joeweiss** [comment]
Actually, now that I think of it, I have an upcoming need for a in-memory file as well. It probably makes sense to have a specific recording class for file objects. Maybe an implementation something like this:

```
import requests
import io

from datetime import datetime
from pprint import pprint
from birdnetlib import RecordingFileObj
from birdnetlib.analyzer import Analyzer

r = requests.get("https://example.com/audio.wav")

analyzer = Analyzer()

with io.BytesIO(r.content) as fileObj:
    recording = RecordingFileObj(
        analyzer,
        fileObj,
        lat=35.4244,
        lon=-120.7463,
        date=datetime(year=2022, month=5, day=10),  # use date or week_48
        min_conf=0.25,
    )
    recording.analyze()
    pprint(recording.detections)
```

Would that work for your needs @MJWeldy?

**Comment by MJWeldy** [comment]
It would, but I think ecologists would end up using librosa or sound files more often, so I would need to verify that the object types were compatible and possibly run a few timing tests.

Do you have any concern about establishing too many class types for specific use cases? In some ways it would be nice to have the analyzer class include some of the recording class methods. Then you could harness tfdataset tools to map across audio files or in memory objects. 

If you'd like to talk it through a bit more let me know.

**Comment by joeweiss** [comment]
> Do you have any concern about establishing too many class types for specific use cases?

Not in the case of Recording. As long as the base class contains 90% of the code, then it makes sense to subclass as needed for different workflows. It's a signal to contributors that there is a preferred way to support new Recordings without resulting to argument soup on a single Recording class.

> In some ways it would be nice to have the analyzer class include some of the recording class methods.

Curious. I would be delighted to take a look at what you mean by that; pseudocode is fine, not an implementation or anything.

**Comment by MJWeldy** [comment]
My thought was that most people use the recording instance once and it
probably doesn't need to be instantiated for each recording. But with that
being said I am not a software developer. Im just an ecologist trying to
hack my way through different analyses :) .

I was thinking something like this

If len(recordings > 1):
   Tf_data = tf.dataset(recordings)

Tf_data.map( analyzer, recordings)

**Comment by joeweiss** [comment]
Ah, I see what you're talking about. Got it. For the projects I'm working on, the recording object (a subclass of Recording in my case) acts as the source of truth for that media while it progresses through the workflows and writes its artifacts to various data stores.

**Comment by joeweiss** [comment]
This has been implemented and released in 0.10.0. 
See https://joeweiss.github.io/birdnetlib/api/#recordingfileobject
